mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -51,6 +58,17 @@
 
     forwardConfigSettings(job);
```

Comment:
@cmathiesen the latest `HiveIcebergInputFormat` has changed substantially. Can you please rebase?

Reply:
> rebase

Shouldn't we be following the [Golden Rule of Rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing#the-golden-rule-of-rebasing) and *not* do this on public branches? It has the potential to cause all kinds of inconsistencies on other people's checkouts. Surely we should be doing merge? It all gets squash merged at the end so having a pristine history isn't worth the downsides of inconsistencies IHMO.

Reply:
I think that either rebasing or merging master into a PR is okay.

As a reviewer, I don't really consider PR branches to be public because github handles force-pushes well. If I have a PR checked out, I also don't mind resetting to the PR's current state because I like keeping history clean.

That said, if you're sharing a PR branch between people that can be disruptive, so I think it is up to the author and collaborators whether to merge or to rebase to stay up to date with master.

Is that reasonable?

Reply:
That sounds reasonable as long as the PR is rebased to master before it is committed. Having a linear history makes things much much easier to track where a change happened.

Reply:
Agreed. We always merge by squashing the entire PR into a commit, so we do get a linear history in master.